### DISCUSSION TITLE
Welcome to gpt-engineer Discussions!

### DISCUSSION THREAD

**Discussion by AntonOsika** [AntonOsika/gpt-engineer#98] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ghost** [comment]
hi

**Reply by syq1990228** [reply]
hi

**Reply by daoudambow** [reply]
slt

**Comment by jbistro** [comment]
Thanks for making this project @AntonOsika

**Comment by sharmaishan2511** [comment]
Hello sir

**Comment by bug-author** [comment]
Hi. A full stack engineer here! curious to see if this would help me solve some multithreading issues and come up with barnd new, creative solutions 😄

**Comment by monkrus** [comment]
Great idea!

**Comment by carreraprogrammer** [comment]
Awesome!!

**Comment by zxaylex** [comment]
Hello!

**Comment by AntonOsika** [comment]
If you're interested in helping:

Check out [this request!](https://github.com/AntonOsika/gpt-engineer/discussions/125)

**Comment by Chals888** [comment]
hi guys, when running my prompt i get the error "openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details." 

I've a free account with openai, and I managed to get api keys, the model is set to GPT-4, how can I fix this? Is this because I need the plus account?

**Reply by whykay-01** [reply]
I am pretty sure that GPT-4 is available only for Plus users. I think that's the main issue a lot of people bump into when it comes to using the gpt-engineer.

**Reply by djpapzin** [reply]
You need to update the main.py file

**Reply by yitron** [reply]
does GPT-Eng only work with GPT4 API Keys? I am a Plus subscriber but as i am in Singapore i do not think we have access to the GPT4 API yet is there anyway around it

**Reply by djpapzin** [reply]
You cna change it to not use GPT4,and you make that change in the main.py file

**Reply by yitron** [reply]
> You cna change it to not use GPT4,and you make that change in the main.py file 

Many thanks!

**Reply by abdullah-2022-skipq** [reply]
@yitron were you able to run it without GPT 4?

**Reply by yitron** [reply]
yes it works some minor errors but it is running on 3.5 Turbo
![Screenshot 2023-06-19 at 12 35 05 PM](https://github.com/AntonOsika/gpt-engineer/assets/53065499/e2124f64-f4c5-48ae-8e34-eef238008677)

**Reply by abdullah-2022-skipq** [reply]
@yitron can you point out those minor changes?

**Reply by abdullah-2022-skipq** [reply]
@yitron Your OpenAI-API Key is visible in the image please hide it

**Reply by yitron** [reply]
hey thanks @abdullah-2022-skipq key revoked.

the minor change is in the main.py file

in line 22 - model: str ="gpt4" to "gpt 3.5"

**Reply by yitron** [reply]
the codes does help you with it 

this was the error from before

rning: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
  warnings.warn(
**Model gpt-4 not available for provided api key reverting to gpt-3.5.turbo.** Sign up for the gpt-4 wait list here: https://openai.com/waitlist/gpt-4-api

**Reply by abdullah-2022-skipq** [reply]
@yitron Thankyou! I'm on Windows so I'm getting some issues in setting up the whole thing lol but I guess after that your solution would work fine for me!

**Comment by yitron** [comment]
Amazing!

**Comment by yitron** [comment]
I’ve not tried it out yet will run it tomorrow I believe it’s a small tweak
as of now

**Comment by yitron** [comment]
home brew is like pip for Mac i code on both windows and Mac

It’s mac’s installer u shud have it in ure working on zsh

**Reply by Mercury888** [reply]
Thank you I didn't see this before I deleted my comment. I have it figured out now. Thanks

**Comment by yogendra-mishra** [comment]
@yitron One basic query - Why GPT engineer needs openAI key?

**Reply by yogendra-mishra** [reply]
One more thing - How is this different from asking chatGPT to do coding and other things. Trying to understand basics first. 

I guess it will save time in copying the code and entering the prompts etc?

**Reply by yogendra-mishra** [reply]
Also since it needs openAPI key, there is a cost associated with using this right (indirectly to OpenAI)?

**Reply by yogendra-mishra** [reply]
It can generate only python as the coding output or any programming language? say build an app in IOS swift. can it do it?

**Reply by kazGuido** [reply]
there is a cost associated =》 yes

**Comment by Kyrpel** [comment]
Hi, how are you? does the tool have the option of continuing the current project and add more instructions?

**Reply by RareMojo** [reply]
If it doesn't, it will soon. I saw the beginnings of this idea when I got started. On my own personal fork, I have this implemented. Does it produce better outputs? Undecided, I will need more time experimenting. So far, I would say it seems to just be a gigantic waste of tokens and other symptoms of re-feeding the data. I could just be a noob after all, too.

**Comment by zuggernautt** [comment]
Hi!

**Comment by jewilhel** [comment]
Is there a way to setup GPT-Engineer on Replit? Could someone provide step by step instructions?
What Replit environment template is required?

**Reply by AntonOsika** [reply]
Great idea. Let’s set this up.

Also: does someone want to create a step in steps.py that creates new projects on replit with their API?

**Reply by jewilhel** [reply]
I hope this can happen soon. I don't want to run GPT-Engineer locally on my laptop. (want to keep my system clean and not run all this experimental stuff on it)

**Comment by jewilhel** [comment]
I followed the setup instructions after setting up a Python template environment in Replit, but got this error when running "make install"

ERROR: File "setup.py" or "setup.cfg" not found. Directory cannot be installed in editable mode: /home/runner/GPT-Engineer/gpt-engineer
(A "pyproject.toml" file was found, but editable mode currently requires a setuptools-based build.)

**Reply by ccromp** [reply]
Getting this error too in replit.

**Comment by efi20232** [comment]
can some one explain me how install it simply.. my git not working so i used download zip i installed it in folder. now what i do what does it mean make install
and where i share my api key of gpt 4

**Comment by Mercury888** [comment]
Hi everyone, Ok, I have done a few different walk through installs and have pretty much everything working except, that I cannot get my API key to work. When I enter it initially I get a green dot and there is no error message but when I go to launch the folder That I created my error message is about the api key, I changed gpt 4 to gpt 3.5-turbo-16k in the main.py folder but I still can't get my key to work even when I use a fresh one. If anyone can give me a general direction on where to go to fix this or how to fix it I would be very grateful, thank you!

**Comment by marrtyw** [comment]
Mostly interested in howto feed some learning materials to gpt4 and make it using in further propts.

**Comment by Nick-Kevin** [comment]
Hi👋! I am front-end developer but I'm still learning and computer science student. 
I am very active with chatGPT so what happens here interests me a lot and I can also give ideas and advice in this community when I have some.

What a great project!! 😉
